Src/FluentAssertions/Common/MemberPath.cs

```diff
@@ -9,56 +11,61 @@ namespace FluentAssertions.Common
     /// </summary>
     internal class MemberPath
     {
-        private readonly Type declaringType;
         private readonly string dottedPath;
+        private readonly string[] segments;
+        private readonly Type reflectedType;
+        private readonly Type declaringType;
 
-        private string[] segments;
+        public MemberPath(IMember member, string parentPath)
+            : this(member.ReflectedType, member.DeclaringType, parentPath.Combine(member.Name))
+        {
+        }
 
-        public MemberPath(Type declaringType, string dottedPath)
+        public MemberPath(Type reflectedType, Type declaringType, string dottedPath)
         {
+            this.reflectedType = reflectedType;
             this.declaringType = declaringType;
             this.dottedPath = dottedPath;
+            segments = dottedPath.Split(new[] { '.', '[', ']' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I seem to recall that `segments` was lazily initialised because not all instances of `MemberPath` needed it to be calculated and it stood out in some profiling.
But my memory may be playing tricks on me.

Reply:
You're right. I need to put that back. I had some intermediate changes that required me to get the segments from the constructor, but I no longer need that.